### DISCUSSION TITLE
Runtime permissions issues on Android side ...

### DISCUSSION THREAD

**Discussion by ndipatri** [Reedyuk/blue-falcon#109] [General]
Hello @Reedyuk .. First of all, thank you so much for all your efforts on this library!  It's much appreciated.  I'm very excited to do BLE KMP.  

The current android version as of 11/13/23 (3fa5be4102c8240e89c2f941a8ae229e489c74dd) seems to have Android runtime permissions issues (at least on Tiramisu).  I was wondering if anyone is addressing that. I'd like to help out, if not.

I'm happy to create an Issue on this and work on it, I just wanted to ping you first and see how you like to do things here... and if you needed some help.

Thanks again!

**Comment by Reedyuk** [comment]
Hey @ndipatri - its my pleasure to make life easier for other fellow developers.

The intention of the library is to pass on the responsibility of permissions to the consuming app. Please look at the KMP example, you will see that an exception is thrown when the permissions are missing.

https://github.com/Reedyuk/blue-falcon/blob/3fa5be4102c8240e89c2f941a8ae229e489c74dd/examples/KotlinMP-Example/androidApp/src/main/java/dev/bluefalcon/kotlinmp_example/android/DevicesView.kt#L37C23-L37C23

Let me know if you need any more details

**Reply by ndipatri** [reply]
Hey there.. Thanks for the response @Reedyuk ,

The [AndroidManifest.xml ](https://github.com/Reedyuk/blue-falcon/blob/master/examples/KotlinMP-Example/androidApp/src/main/AndroidManifest.xml#L3)for the KotlinKMP-Example doesn't declare the BLUETOOTH_SCAN permission so it's not getting beyond that first check... at least from what I can tell.

**Reply by Reedyuk** [reply]
it is recommended you put that permission in your Android app manifest and not in the library one

**Reply by ndipatri** [reply]
Right.. I got that.  The [AndroidManifest.xml ](https://github.com/Reedyuk/blue-falcon/blob/master/examples/KotlinMP-Example/androidApp/src/main/AndroidManifest.xml#L3) that I referenced above is in the example Android app that you created - not in the library.  It also seems to be missing the ACCESS_FINE_LOCATION check as well...  So this issue seems to just be related to the example app, not the library itself.

**Reply by Reedyuk** [reply]
yes that was a mistake, i had rushed the example out for another person. Please create a pr with the change if you want.

**Reply by ndipatri** [reply]
oh awesome! I would love the chance to contribute to this awesome library.. It might take a few days... but I'm ON IT! :-)

**Reply by Reedyuk** [reply]
@ndipatri fork my repo and then do the change and create a pull request on your own repo, it will point the source branch to this one and I can review. This is how open source projects work.

**Reply by ndipatri** [reply]
heh.. It's been a while and i forgot that I hadn't forked first.. derp.. Thanks for the heads up!

**Comment by Reedyuk** [comment]
This is now solved.
